### PR TITLE
expose cookie validity as config - fix #2157

### LINKED ISSUE #2157: cookie lifetime
I use skipper to do OIDC pre-auth with the filter oauthOidcAllClaims. I noticed that the skipper cookie has a lifetime of 1 hour. For a Single-Page-Application the following situation happens: If a user let the site open for over 1 hour skipper tries to refresh the cookie and does a redirect to the auth-service. For normal apps/sites this works (http-requests). But for the XHR-Requests of javascript it doesn't work. In the end this leads to a failure on the Single-Page-Application. Only a full reload of the page helps to solve it. Are there some ideas how to fix that? Is it e.g. possible to set a longer lifetime for the skipper cookie? Or are there some parameters/filters to avoid such a situation?

I would prefer to set the cookie lifetime globally via flag.

### PR DESCRIPTION
**Pull request by szuecs**
expose cookie validity as config
fix https://github.com/zalando/skipper/issues/2157

TODO:
- [x] tests
- [x] exp integration (checked because done already)

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

File: config/config.go [hunk 1]
```diff
@@ -203,6 +203,7 @@ type Config struct {
 	Oauth2TokenCookieName           string        `yaml:"oauth2-token-cookie-name"`
 	WebhookTimeout                  time.Duration `yaml:"webhook-timeout"`
 	OidcSecretsFile                 string        `yaml:"oidc-secrets-file"`
+	OIDCCookieValidity              time.Duration `yaml:"oidc-cookie-validity"`
```
Comment (AlexanderYastrebov) [inline config/config.go:206]: Shouldn't we add a flag for it?

Comment (szuecs) [reply config/config.go:206]: oops, yes

Comment (szuecs) [reply config/config.go:206]: fixed

**Comment by MustafaSaber** [comment]
👍

**Comment by szuecs** [comment]
:+1:

**Pull request merged**